src/mugs.js

```diff
@@ -295,7 +295,45 @@ define([
             rawControlXML: {
                 presence: 'optional',
                 lstring: 'Raw XML'
-            }
+            },
+            dataParent: {
+                lstring: 'Data Parent',
+                visibility: function(mug) {
+                    function recFunc(mug) {
+                        if (!mug) {
+                            return true;
+                        } else if (mug.__className === 'Repeat' ||
+                                   mug.__className === "Item") {
```

Comment:
A repeat node can have a different data parent, but cannot be a data parent. Also children of the repeat node cannot have different data parents. This is checked in the validation function. There are also two tests:  

https://github.com/dimagi/Vellum/pull/340/files#diff-597c6d5c9e1efe9adf37cd233ade5331R32

https://github.com/dimagi/Vellum/pull/340/files#diff-597c6d5c9e1efe9adf37cd233ade5331R80